frontends/uhdm/UhdmAst.cc

```diff
@@ -248,8 +248,55 @@ static void add_or_replace_child(AST::AstNode* parent, AST::AstNode* child) {
 			*it = child;
 			return;
 		}
+	} else if (child->type == AST::AST_INITIAL) {
+		// Special case for initials
+		// Ensure that there is only one AST_INITIAL in the design
+		// And there is only one AST_BLOCK inside that initial
+		// Copy nodes from child initial to parent initial
+		auto initial_node_it = find_if(parent->children.begin(), parent->children.end(),
+			[] (AST::AstNode *node) {return (node->type == AST::AST_INITIAL);} );
+		if (initial_node_it != parent->children.end()) {
+			AST::AstNode* initial_node = *initial_node_it;
+			auto block_node_it = find_if(initial_node->children.begin(), initial_node->children.end(),
+				[] (AST::AstNode *node) {return (node->type == AST::AST_BLOCK);} );
+			if (block_node_it != initial_node->children.end()) {
+				AST::AstNode* block_node = *block_node_it;
+				auto child_block_node_it = find_if(child->children.begin(), child->children.end(),
+					[] (AST::AstNode *node) {return (node->type == AST::AST_BLOCK);} );
+				if (child_block_node_it != child->children.end()) {
+					// Place the contents of child initial node inside the parent initial
+					// But place the contents of child block node inside parent block
+					AST::AstNode* child_block_node = *child_block_node_it;
+					for (auto initial_child : child->children) {
+						if (initial_child == child_block_node) {
+							for (auto child_block_child : child_block_node->children)
+								block_node->children.push_back(child_block_child);
+						} else {
+							initial_node->children.push_back(initial_child);
+						}
+					}
+				} else {
+					// AST_BLOCK in child does not exist, place the contents
+					// of child initial node inside the parent
+					for (auto initial_child : child->children)
+						initial_node->children.push_back(initial_child);
+				}
+			} else {
+				// AST_BLOCK in parent does not exist, place the contents
+				// of child initial node inside the parent
+				for (auto initial_child : child->children)
+					initial_node->children.push_back(initial_child);
```

Comment:
Are this cases even possible? This AST_BLOCK should be created when creating AST_INITIAL, so I think we should check/assert that in ``child``/``parent`` node there is AST_BLOCK as a first children rather then this.

Reply:
Added checks and throwing errors when conditions aren't met